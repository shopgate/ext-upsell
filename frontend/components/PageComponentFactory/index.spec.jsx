import React from 'react';
import { mount } from 'enzyme';
import pageComponentFactory from './index';

describe('pageComponentFactory', () => {
  // eslint-disable-next-line require-jsdoc, react/prop-types
  const View = ({ children }) => (<div id="view">{children}</div>);
  // eslint-disable-next-line require-jsdoc, react/prop-types, no-shadow
  const ComponentProto = ({ View }) => (<View><div id="component">Hello world</div></View>);

  it('should render a component in a view', () => {
    const ComponentClass = pageComponentFactory({
      View,
      ComponentProto,
    });
    const component = mount(<ComponentClass />);

    expect(component.exists('#view')).toBe(true);
    expect(component.exists('#component')).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
