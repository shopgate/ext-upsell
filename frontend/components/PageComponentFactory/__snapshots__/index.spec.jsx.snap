// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pageComponentFactory should render a component in a view 1`] = `
<_class>
  <ComponentProto
    View={[Function]}
  >
    <View>
      <div
        id="view"
      >
        <div
          id="component"
        >
          Hello world
        </div>
      </div>
    </View>
  </ComponentProto>
</_class>
`;
